### PR TITLE
Round three of erroneous shard removal

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Supersedes #468

This switches to a rolling window approach; removing only the shards that have not been built in a span of time (1 year to start).  This approach will only remove "inactive" shards, and give additional times for more active shards to setup periodic builds.

### Setting up Periodic builds
* TravisCI - https://docs.travis-ci.com/user/cron-jobs/
* GH Actions - https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#scheduled-events-schedule
* CircleCI - https://circleci.com/docs/2.0/configuration-reference/#schedule
* DroneCI - https://blog.drone.io/drone-1/#cron-scheduling
* GitLab - https://docs.gitlab.com/ee/user/project/pipelines/schedules.html

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -229,33 +206,25 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [syringe](https://github.com/Bonemind/syringe) - A simple and basic dependency injection shard for crystal
 
 ## Development Tools
```
Comment (veelenga) [inline README.md]: Please remove an empty section and clean TOC

File: README.md [hunk 2]
```diff
@@ -276,30 +243,23 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [glove](https://github.com/ddfreyne/glove) - A library for gaming development
 
 ## HTML Builders
```
Comment (veelenga) [inline README.md]: Ditto

File: util/cli.cr [hunk 3]
```diff
@@ -5,6 +5,9 @@ require "http/client"
 require "colorize"
 require "option_parser"
 
+# A shard must have been built sometime in this span in order to be considered active.
+private TIME_WINDOW = (30 * 12).days # 12 months
```
Comment (veelenga) [inline util/cli.cr]: what do you think about more accurate calculations? Maybe using `1.year` or something.

File: README.md (hunk 1 above)
Comment (Blacksmoke16) [reply README.md]: I'll do that before merging.  So that the sections are still there if people fix their shard.

**Comment by Blacksmoke16** [comment]
## Shards To Be Removed

The following shards have not had a build in the last 1 year.  Please setup a periodic build if your project is still maintained, otherwise just ignore this.

https://github.com/0x73/hoop - @0x73
https://github.com/CodeSteak/Nuummite - @CodeSteak
https://github.com/DanSnow/crystal-binary_parser - @DanSnow
https://github.com/EvanHahn/crystal-helmet - @EvanHahn
https://github.com/SuperPaintman/request-id - @SuperPaintman
https://github.com/SuperPaintman/crystal-ctags - @SuperPaintman
https://github.com/SuperPaintman/serve - @SuperPaintman
https://github.com/SuperPaintman/response-time - @SuperPaintman
https://github.com/SuperPaintman/ms - @SuperPaintman
https://github.com/TPei/progress_bar.cr - @TPei
https://github.com/TamasSzekeres/asound-cr - @TamasSzekeres
https://github.com/TamasSzekeres/cairo-cr - @TamasSzekeres
https://github.com/TamasSzekeres/x11-cr - @TamasSzekeres
https://github.com/TobiasGSmollett/hash_ring - @TobiasGSmollett
https://github.com/TobiasGSmollett/fluent-logger-crystal - @TobiasGSmollett
https://github.com/abvdasker/crystal-linked-list - @abvdasker
https://github.com/akiicat/m3u8 - @akiicat
https://github.com/anicholson/gopher.cr - @anicholson
https://github.com/beanieboi/aws-signer.cr - @beanieboi
https://github.com/benoist/terminal_table.cr - @benoist
https://github.com/blocknotes/kemal-rest-api - @blocknotes
https://github.com/comandeo/crystal-memcached - @comandeo
https://github.com/comandeo/syslog.cr - @comandeo
https://github.com/crystal-lang/html_builder - @crystal-lang
https://github.com/danielpclark/keyer_cr - @danielpclark
https://github.com/dhruvrajvanshi/crystal-futures - @dhruvrajvanshi
https://github.com/diggersheep/warden - @diggersheep
https://github.com/dlanileonardo/sendgrid.cr - @dlanileonardo
https://github.com/emancu/crotest - @emancu
https://gitlab.com/ezrast/crometheus - @ezrast
https://github.com/faustinoaq/watcher - @faustinoaq
https://github.com/faustinoaq/sentry-run - @faustinoaq
https://github.com/faustinoaq/kemal-watcher - @faustinoaq
https://github.com/fazibear/tensorflow.cr - @fazibear
https://github.com/felipeelias/crystal-github - @felipeelias
https://github.com/gdonald/blackjack-cr - @gdonald
https://github.com/geppetto-apps/crow - @geppetto-apps
https://github.com/gewo/bugsnag.cr - @gewo
https://github.com/icyleaf/wasp - @icyleaf
https://github.com/isaced/pangu.cr - @isaced
https://github.com/izniburak/denetmen - @izniburak
https://github.com/izniburak/query-builder - @izniburak
https://github.com/izniburak/turkish-number - @izniburak
https://github.com/jeromegn/bson.cr - @jeromegn
https://github.com/johnjansen/clamp - @johnjansen
https://github.com/kemalyst/kemalyst-generator - @kemalyst
https://github.com/luislavena/radix - @luislavena
https://github.com/maiha/redis-tsv.cr - @maiha
https://github.com/maiha/hq.cr - @maiha
https://github.com/marceloboeira/bojack - @marceloboeira
https://github.com/measurechina/eventql-crystal - @measurechina
https://github.com/metaware/flipper - @metaware
https://github.com/mjago/soundfile - @mjago
https://github.com/mosop/cli - @mosop
https://github.com/mosop/optarg - @mosop
https://github.com/obsidian/oak - @obsidian
https://github.com/philnash/envyable.cr - @philnash
https://github.com/phoffer/inflector.cr - @phoffer
https://github.com/rhysd/Crisp - @rhysd
https://github.com/rogeriozambon/http-protection - @rogeriozambon
https://github.com/sdogruyol/tren - @sdogruyol
https://github.com/sdogruyol/cryload - @sdogruyol
https://github.com/splattael/libnotify.cr - @splattael
https://github.com/topaz-crystal/topaz - @topaz-crystal
https://github.com/unn4m3d/cppize - @unn4m3d
https://github.com/veelenga/play-crystal.el - @veelenga
https://github.com/vjdhama/crystal-cookbook - @vjdhama
https://github.com/westonganger/form_builder.cr - @westonganger
https://github.com/z64/openssl.cr - @z64

**Comment by abvdasker** [comment]
abvdasker/crystal-linked-list had new commits about a month ago. Would it
be possible to keep it?

**Comment by luislavena** [comment]
@Blacksmoke16 feel free to remove Radix from the list, however the library is feature-complete (except for one bug) and setting up an periodic build seems a bit too much waste of resources.

**Comment by gdonald** [comment]
@Blacksmoke16 feel free to remove

https://github.com/gdonald/blackjack-cr - @gdonald 

from the list.  It is feature-complete and IMHO setting up a periodic build is a waste of resources.

**Comment by Blacksmoke16** [comment]
@abvdasker Ah, it looks like the project was using CircleCI 1.0, which cause CI to not run for those commits.

> Service alert: Your project references CircleCI 1.0 or it has no configuration. CircleCI 1.0 and projects without configuration files are no longer supported. You must update your project to use CircleCI 2.0 configuration to continue. Learn more.

Looks like you just need to update the configuration for 2.0 and it'll be good.  https://circleci.com/sunset1-0/

@luislavena @gdonald As you wish.  For how trivial it would be to setup a weekly build is your call.

**Comment by luislavena** [comment]
>  For how trivial it would be to setup a weekly build is your call.

@Blacksmoke16 would accept a PR to setup that with either Travis (currently in use) or GitHub Actions (if Crystal works with it). But I don't have the bandwidth right now to tackle any OSS projects.

Cheers.

**Comment by Blacksmoke16** [comment]
@luislavena Setting it up on Travis would take 1min as it's just clicking a few buttons.  https://docs.travis-ci.com/user/cron-jobs/

**Comment by danielpclark** [comment]
My old shard/project is listed here.  It's been a while since I've worked with Crystal.  Is there a binary I need to build for those who use my library?  What does "removing the shard" mean… does that mean remove from a publicly listed directory of packages, or maybe a stored binary package?

**Comment by icyleaf** [comment]
@Blacksmoke16  feel free to remove.

https://github.com/icyleaf/wasp - @icyleaf

**Comment by philnash** [comment]
Thanks, cron set up to build https://github.com/philnash/envyable.cr periodically. 👍

**Comment by Blacksmoke16** [comment]
@danielpclark It just means removing the shard from [the list](https://github.com/veelenga/awesome-crystal#awesome-crystal).

If you're still maintaining it just setup a periodic build for it and it'll be fine.

See [this](https://github.com/veelenga/awesome-crystal/pull/457#issuecomment-552223501) comment for the background context.

**Comment by TPei** [comment]
https://github.com/TPei/progress_bar.cr still green with 0.33.0, will set up periodic builds some time this week

**Comment by abvdasker** [comment]
Ok, abvdasker/crystal-linked-list has been updated to CircleCI 2.0 . Can we
keep this repo on the list?

**Comment by luislavena** [comment]
@Blacksmoke16 added weekly builds for Radix and triggered a manual one. Project still compatible.

File: README.md [hunk 4]
```diff
@@ -470,7 +408,6 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [Water](https://github.com/shootingfly/water) - A library for writing HTML in plain Crystal
 
 ## Testing
- * [crotest](https://github.com/emancu/crotest) - A tiny and simple test framework
```
Comment (emancu) [inline README.md]: I don't understand the criteria.
The shard is feature complete and stable, why remove it?

**Comment by splattael** [comment]
@Blacksmoke16 Thanks for taking care :bowing_man: 

I've updated https://github.com/splattael/libnotify.cr to make sure it compiles on Crystal `0.33.0` and released a new version :+1: 

Please don't remove this shard from the list :pray:

File: README.md (hunk 4 above)
Comment (Blacksmoke16) [reply README.md]: See https://github.com/veelenga/awesome-crystal/pull/457#issuecomment-552223501

**Comment by abvdasker** [comment]
@Blacksmoke16 https://github.com/abvdasker/crystal-linked-list  has been updated to CircleCI 2.0 and has generated new builds.

**Comment by Blacksmoke16** [comment]
@abvdasker Yup I saw.  I'll update the list at some point this weekend, then once again next weekend.

File: README.md (hunk 4 above)
Comment (emancu) [reply README.md]: Ok. I added weekly builds, and updated to crystal 0.33!

**Comment by beanieboi** [comment]
I updated https://github.com/beanieboi/aws-signer.cr to work with the latest crystal version. You can keep it in the list

**Comment by neovintage** [comment]
I updated https://github.com/neovintage/accord with github actions. It'll kick off a monthly build with the latest crystal version. 😄

**Comment by rhysd** [comment]
Hello, I fixed [Crisp](https://github.com/rhysd/crisp) to make it compatible with the latest Crystal v0.33.0 at https://github.com/rhysd/Crisp/commit/0b970e875586db261e42a49a9f9d8a1a013f1e3c.

**Comment by rhysd** [comment]
https://github.com/rhysd/Crisp/commit/b7e05f1094bbac3c8c199b4fe655dc6f8e5970b0

I've setup GitHub Actions for checking build and tests periodically.

**Comment by westonganger** [comment]
I've added periodic builds for `form_builder.cr`

**Comment by veelenga** [comment]
@Blacksmoke16 please rebase this work. We are ready to merge it.

**Comment by dlanileonardo** [comment]
Hi @Blacksmoke16 

- [x] I Fixed - [sendgrid.cr](https://github.com/dlanileonardo/sendgrid.cr) 
- [x] Update to Crystal v0.34.1 - [commit](https://github.com/dlanileonardo/sendgrid.cr/commit/59b61f06e94252f439198fdbeed7a82be09153fa) 
- [x] Setup Github Actions, to build periodically - [Runs](https://github.com/dlanileonardo/sendgrid.cr/runs/)

**Comment by veelenga** [comment]
Thanks everyone for your effort and understanding.

**Pull request merged**